alpa/pipeline_parallel/pipeshard_executable.py

```diff
@@ -1311,7 +371,7 @@ def __init__(self, worker: MeshHostWorker, uuid: int,
         self.global_buffers = worker.buffers
         self.acc_grad_buffers = {}
         self.acc_in_uuids = [list(uuids) for uuids in list(acc_local_uuids)]
-        self.acc_out_uuids = acc_out_uuids
+        self.acc_out_uuids = [list(uuids) for uuids in list(acc_out_uuids)]
```

Comment:
You often convert a numpy array to a list by calling `list()`. Most of them are not necessary and I deleted a lot of this pattern last time. Try to remove them and work directly with numpy indexing.

Reply:
Currently in PipeshardExecutable there is almost no `list(XXX)`. I'll check the PipelineInstEmitter.